src/common/src/hash/key.rs

```diff
@@ -213,13 +217,110 @@ pub struct FixedSizeKey<const N: usize> {
     null_bitmap: NullBitmap,
 }
 
+/// Able to store ~8 i64s as group keys.
+#[repr(transparent)]
+#[derive(Clone, Debug, PartialEq)]
+struct SerializedKeyBuffer {
+    inner: SmallVec<[u8; 32]>,
```

Comment:
This seems too large for the stack size. Would you please test `size_of` for this type? I guess with `16` we're not worse than directly using a `Vec`.

Reply:
Minimum is stack space is 24 bytes, since internally they use a `union` type, which is baseline size for the data segment of the smallvec. The data segment occupies 16 bytes. The capacity would occupy 8 bytes. In total that would be `16+8=24`.

I guess best case scenario is using 16 bytes on stack (i.e. `SmallVec<[u8; 16]>`).

The benchmark for that shows poor performance, with around `16-20%` regression compared to `main`:
```
vec ser 1024 rows, mix fixed and not fixed, case 2 KeySerialized, Pr[null]=0
                        time:   [190.78 µs 191.15 µs 191.55 µs]
                        change: [+15.385% +15.794% +16.197%] (p = 0.00 < 0.05)
                        Performance has regressed.
Found 3 outliers among 100 measurements (3.00%)
  2 (2.00%) high mild
  1 (1.00%) high severe

vec ser 1024 rows, mix fixed and not fixed, case 2 KeySerialized, Pr[null]=0.01
                        time:   [173.13 µs 173.83 µs 174.57 µs]
                        change: [+17.580% +18.163% +18.725%] (p = 0.00 < 0.05)
                        Performance has regressed.
Found 7 outliers among 100 measurements (7.00%)
  7 (7.00%) high mild

vec ser 1024 rows, mix fixed and not fixed, case 2 KeySerialized, Pr[null]=0.1
                        time:   [155.28 µs 155.66 µs 156.05 µs]
                        change: [+19.918% +20.242% +20.582%] (p = 0.00 < 0.05)
                        Performance has regressed.
Found 10 outliers among 100 measurements (10.00%)
  6 (6.00%) low mild
  3 (3.00%) high mild
  1 (1.00%) high severe
```

Reply:
See: https://github.com/risingwavelabs/risingwave/blob/292d4acd33f7237d6f15445a2157ef68d4782082/src/common/src/hash/key.rs#L229-L249